src/util/promise.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * This function emulates the Promise.allSettled()
+ *
+ * @param proms
+ */
+export function allSettled(proms: Array<Promise<any>>) {
```

Comment:
Actually, the more correct typedef would be: 

```ts
export function allSettled(proms: Array<Promise<any>>): Array<{ status: "fulfilled", value: any } | { status: "rejected", reason: Error }> {
```

But then, clients need to manually check `status == "fulfilled"` in order for TS to infer that `value` would exist (or `reason` otherwise)

Reply:
I think it should be:

```ts
export function allSettled(proms: Array<Promise<any>>): Promise<Array<{ value: any, status: string } | { reason: Error, status: string }>> {
```